tests/unit/actions/test_create.py

```diff
@@ -103,3 +103,20 @@ def test_run_create_bails_if_repository_does_not_match():
             remote_path=None,
         )
     )
+
+
+def test_create_borgmatic_manifest_creates_manifest_file():
+    flexmock(module.os.path).should_receive('expanduser').and_return('/home/user')
+    flexmock(module.os.path).should_receive('join').and_return('/home/user/bootstrap/manifest.json')
+    flexmock(module.os.path).should_receive('exists').and_return(False)
+    flexmock(module.os).should_receive('makedirs').and_return(True)
+
+    flexmock(module.json).should_receive('dump').and_return(True)
+
+    module.create_borgmatic_manifest({}, 'test.yaml', False)
```

Comment:
What difference would that make test-wise? Only this line is concerned with the source directory right - 

```python
    borgmatic_source_directory = location.get(
        'borgmatic_source_directory', DEFAULT_BORGMATIC_SOURCE_DIRECTORY
    )
```

So should I check if I do give it a location, it should use that instead of the DEFAULT directory? I though `.get()` won't induce any unexpected behaviour so I didn't add it.

Reply:
Yeah, it's less about testing that `.get()` does what it's supposed to, but more about testing that the function as a whole (regardless of its implementation) adheres to its contract of honoring the `borgmatic_source_directory` option. If someone comes along later and accidentally deletes that `.get()` line or changes it to hard-code `DEFAULT_BORGMATIC_SOURCE_DIRECTORY`, you ideally want a test failing loudly (and clearly). Getting on my [soapbox](https://idioms.thefreedictionary.com/soapbox) for a moment: Tests should be about defining and enforcing the contract that a unit under test has with its caller. If a part of the contract _isn't_ defined in the tests, then it's not part of the contract IMO!

Having said all that, I don't actually feel strongly about this particular example, so follow your heart. :smile:

Reply:
> Tests should be about defining and enforcing the contract that a unit under test has with its caller. If a part of the contract isn't defined in the tests, then it's not part of the contract.

True, will remember this quote when writing tests in the future 😄